### PR TITLE
Add Bearer Authentication

### PR DESCRIPTION
**Pull request by jmrtnz94**

### CODE REVIEW

**Comment by jmrtnz94** [comment]
I can work on that tomorrow. Thanks for the quick reply!

File: src/ApiFirstMediatR.Generator/Templates/Controller.cs.liquid [hunk 1]
```diff
@@ -23,6 +23,9 @@ namespace {{ controller_namespace }}
         /// <param name="{{ parameter.parameter_name }}">{{ parameter.description | join: "<br/>" }}</param>
         {%- endfor %}
         /// <returns>{{ endpoint.response.description | join: "<br/>" }}</returns>
+        {%- for security in endpoint.securities %}
+        {% if security.type == "Http" %}[Microsoft.AspNetCore.Authorization.Authorize{% if security.scopes.size != 0 %}(Policy = "{{ security.scopes | first }}"){% endif %}]{% endif %}
+        {%- endfor %}
```
[the remark this answers is not in the document]
Comment (jmrtnz94) [reply src/ApiFirstMediatR.Generator/Templates/Controller.cs.liquid:28]: This is the outcome of that change

image

**Comment by jmrtnz94** [comment]
I've pushed new commits, let me know what you think.

An Endpoint now has a Security object.

The Security object has a list of Polices.

**Comment by jmrtnz94** [comment]
@alexcampana, what are your thoughts on this commit? Should we update the openapi spec file to have an example of using a Bearer Authentication?

https://github.com/alexcampana/ApiFirstMediatR/pull/60/commits/4c3c051d6730e7d2c9015cb26b58516b4a8414ed

**Comment by jmrtnz94** [comment]
I've added unit test for the Security mapper, and updated the Endpoint mapper unit test.

File: src/ApiFirstMediatR.Generator/Templates/Controller.cs.liquid (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexcampana) [reply src/ApiFirstMediatR.Generator/Templates/Controller.cs.liquid:28]: I'm really not sure why that change was working on my machine before. But I'm definitely seeing the same thing your are now.

**Comment by alexcampana** [comment]
> @alexcampana, what are your thoughts on this commit? Should we update the openapi spec file to have an example of using a Bearer Authentication?
> 
> [4c3c051](https://github.com/alexcampana/ApiFirstMediatR/commit/4c3c051d6730e7d2c9015cb26b58516b4a8414ed)

I definitely like it, at some point I'd like to use that policy check in the actual integration tests, but definitely not a problem that needs to be solved now.

**Comment by alexcampana** [comment]
Looks good, I'll get a new nuget package out shortly!

**Review by alexcampana — APPROVED** [review]

**Pull request merged**